tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -390,7 +390,7 @@ def test_flyte_file_in_dataclass():
     @dataclass
     class InnerFileStruct(DataClassJsonMixin):
         a: FlyteFile
-        b: PNGImageFile
+        b: FlyteFile
```

Comment:
why do we change the type to `FlyteFile` from `PNGImageFile` here?

Reply:
Because it is declared as PNGImageFIle, it will be checked whether the declared file type is consistent with the actual file format. However, in the current unit test, there is no actual file, which will cause a FileNotFoundError.
like this
![Screenshot 2023-11-07 at 9 03 06 AM](https://github.com/flyteorg/flytekit/assets/14800485/2ef584ef-7450-4cca-b064-cb1715a00250)

https://github.com/jasonlai1218/flytekit/blob/add-validation-flyteformat-type/tests/flytekit/unit/core/test_type_hints.py#L426-L430

Reply:
![Screenshot 2023-11-07 at 10 03 44 AM](https://github.com/flyteorg/flytekit/assets/14800485/e91199d7-b830-4d95-93b1-66eedf3cc796)